### PR TITLE
cmd/txtar-addmod: allow stdout destination

### PR DESCRIPTION
**Pull request by rogpeppe**
This makes it easier to produce inline module definitions for
inclusion in txtar files.

### CODE REVIEW

File: cmd/txtar-addmod/addmod.go [hunk 1]
```diff
@@ -36,6 +36,22 @@ import (
 func usage() {
 	fmt.Fprintf(os.Stderr, "usage: txtar-addmod dir path@version...\n")
 	flag.PrintDefaults()
+
+	fmt.Fprintf(os.Stderr, `
+The txtar-addmod command adds a module as a txtar archive to the a
+testdata module directory as understood by the goproxytest package
+(see https://godoc.org/github.com/rogpeppe/go-internal/goproxytest).
+
+The dir argument names to directory to add the module to. If dir is "-",
```
Comment (myitcv) [inline cmd/txtar-addmod/addmod.go]: "names _the_ directory"

Comment (myitcv) [reply cmd/txtar-addmod/addmod.go]: Also, I'm aware of the use of `-` as stdin, but stdout?

File: cmd/txtar-addmod/addmod.go [hunk 2]
```diff
@@ -36,6 +36,22 @@ import (
...
+the result will instead be written to the standard output in a form
+suitable for embedding directly into a testscript txtar file, with each
+file prefixed with the ".gomodproxy" directory.
+
+In general, txtar-addmod is intended to be used only for very small
+modules - we do not want to check very large files into testdata/mod.
```
Comment (myitcv) [inline cmd/txtar-addmod/addmod.go]: I ended up hacking together a Go script that produces skeleton mod definitions. i.e. the initial `module` line followed by the `.mod` and `.info`

Might be worth providing this option via a flag?

File: cmd/txtar-addmod/addmod.go [hunk 3]
```diff
@@ -133,12 +149,19 @@ func main1() int {
 		if !strings.Contains(arg, "@") {
 			title += "@" + vers
 		}
-		a.Comment = []byte(fmt.Sprintf("module %s\n\n", title))
+		dir = filepath.Clean(dir)
+		modDir := strings.Replace(path, "/", "_", -1)+"_"+vers
```
Comment (myitcv) [inline cmd/txtar-addmod/addmod.go]: I wonder whether we should extract this "logic"; it's repeated in `goproxytest`.

File: cmd/txtar-addmod/addmod.go [hunk 4]
```diff
@@ -36,6 +36,22 @@ import (
 func usage() {
 	fmt.Fprintf(os.Stderr, "usage: txtar-addmod dir path@version...\n")
 	flag.PrintDefaults()
+
+	fmt.Fprintf(os.Stderr, `
+The txtar-addmod command adds a module as a txtar archive to the a
```
Comment (myitcv) [inline cmd/txtar-addmod/addmod.go]: s/to the a testdata/to the testdata/

File: cmd/txtar-addmod/addmod.go (hunk 1 above)
Comment (rogpeppe) [reply cmd/txtar-addmod/addmod.go]: Yeah, it's not uncommon (the `tar` command uses that convention for example).

File: cmd/txtar-addmod/addmod.go (hunk 2 above)
Comment (rogpeppe) [reply cmd/txtar-addmod/addmod.go]: Probably in another PR. I'd like to know exactly how you use this and what semantics you chose though, so I don't need to think that hard about it :)

Comment (myitcv) [reply cmd/txtar-addmod/addmod.go]: Yes, agreed. Doing this in another PR makes more sense.

File: cmd/txtar-addmod/addmod.go (hunk 3 above)
Comment (rogpeppe) [reply cmd/txtar-addmod/addmod.go]: Yeah, probably. Perhaps not right now - this is an existing issue and it's not really a pressing one (we can't change the convention anyway, as too many existing txtar files use it)

File: cmd/txtar-addmod/addmod.go (hunk 4 above)
Comment (rogpeppe) [reply cmd/txtar-addmod/addmod.go]: done

**Pull request merged**